### PR TITLE
Use wfl bot and process orders

### PR DESCRIPTION
**Pull request by TNagaraja**
Issue #8

### CODE REVIEW

File: wfl-slack-app/src/wfl-bot.js
Comment (TNagaraja) [inline wfl-slack-app/src/wfl-bot.js]: Note that these are not slash commands. They are just normal messages since the bot is listening to the messages in the channel.

**Review by enricoferrari — APPROVED** [review]
I am not too familiar with these APIs so I don't have much feedback here.  I will probably have more feedback on this after I get a chance to play with it. The code looks good to me.

File: wfl-slack-app/src/slack-api.js
Comment (JChapman202) [inline wfl-slack-app/src/slack-api.js]: Why not return the entire user object from this API?

File: wfl-slack-app/src/slack-api.js
Comment (JChapman202) [inline wfl-slack-app/src/slack-api.js:11]: I'm curious what your thoughts are regarding isomophic-fetch?  How did you choose this library for REST API processing?

File: wfl-slack-app/src/wfl-api.js
Comment (JChapman202) [inline wfl-slack-app/src/wfl-api.js:28]: Why did we choose to use both fetch and request for REST APIs?

File: wfl-slack-app/src/wfl-api.js
Comment (JChapman202) [inline wfl-slack-app/src/wfl-api.js:32]: Along those lines, Did you look at all at REST API clients which support Siren natively?  I wrote on ea long time ago called super-siren (https://www.npmjs.com/package/super-siren).  I'm not 100% sure it's still compatible with node 6.2 but it may be worth taking a look at?

File: wfl-slack-app/src/wfl-api.js
Comment (JChapman202) [inline wfl-slack-app/src/wfl-api.js:58]: why did you choose to put some functions inline in the exports and some outside?

File: wfl-slack-app/src/wfl-api.js
Comment (JChapman202) [inline wfl-slack-app/src/wfl-api.js:60]: I think this is where we'll have to make a decision.  Hypermedia and proper Siren understanding should help a bit here, as it completely removes the need to combine URLs like this.  But at the same time, I want to have something that fully works before you are finished.  I'm thinking it'll make sense to get this working as is, get it merged, then dig into leveraging hypermedia from a client perspective.

File: wfl-slack-app/src/wfl-bot.js
Comment (JChapman202) [reply wfl-slack-app/src/wfl-bot.js]: Using indexOf finds if that string is anywhere in the text.  Is that what you wanted?  Or did you instead want to have the message start with that text?  You could also use `startsWith` if that's what you want.

File: wfl-slack-app/src/handle-wfl-request.js
Comment (JChapman202) [inline wfl-slack-app/src/handle-wfl-request.js:72]: While I don't think our main effort was learning javascript.  I think after this PR is merged it would make sense to discuss promises and how it can simplify some of this programming paradigm.  I know you used them a little bit with your fetch code.

**Review by JChapman202 — APPROVED** [review]
I don't think there is anything we have to change here to merge this.  However, I do think there are a number of topics that would be worth a conversation, and we could potentially iterate on them for a 2nd version of the bot.

Good work here.  There is a lot of capability built into this bot.  You clearly figured out how to solve a lot of problems using tools you were not completely familiar with and made it function.  Nice work!

File: wfl-slack-app/src/wfl-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/wfl-api.js:58]: I was referencing some exported functions from other exported functions. Was not sure how to do that without defining the referenced functions outside of module.exports.

File: wfl-slack-app/src/slack-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/slack-api.js:11]: I used isomorphic-fetch in `slack-api.js` because that's how we interacted with slack from the work from home app and I could reuse this bit of code. I could use request here as well.

File: wfl-slack-app/src/slack-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/slack-api.js]: Good point. I can rework this to return the entire user object.

File: wfl-slack-app/src/wfl-bot.js
Comment (TNagaraja) [reply wfl-slack-app/src/wfl-bot.js]: Used startsWith instead

File: wfl-slack-app/src/wfl-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/wfl-api.js:28]: I mentioned this in the other comment about fetch, but I did this just because I wanted to be able to reuse the code from the wfh app when calling to the slack api. I could convert the slack-api.js code to use request instead if you think that would make for better uniformity across classes.

File: wfl-slack-app/src/wfl-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/wfl-api.js:32]: It would be nice to be able to use the actions we created in the hypermedia profiles for this project. I didn't even realize there were references we could use that could do this. Considering the restricted time frame I have, I think it might be a good idea to merge this PR when any pressing requested changes that might exist are handled and open a new PR to use this library.

File: wfl-slack-app/src/wfl-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/wfl-api.js:60]: I agree with this. It would be nice to see hypermedia leveraged both at the web api we created and in the slack app that sends request to it as well. I also agree that it may be best to get this PR merged first so that we have something that works going before my internship ends.

File: wfl-slack-app/src/handle-wfl-request.js
Comment (TNagaraja) [reply wfl-slack-app/src/handle-wfl-request.js:72]: I have some limited experience with promises at the moment. Is there an advantage to using promises over passing callback functions into these methods? I liked being able to pass in various different callback functions to the functions accepting them.

File: wfl-slack-app/src/slack-api.js
Comment (TNagaraja) [reply wfl-slack-app/src/slack-api.js]: Changed to return entire user object in latest code

File: wfl-slack-app/src/wfl-api.js
Comment (JChapman202) [reply wfl-slack-app/src/wfl-api.js:58]: I think that's fine, but I would then just make all functions defined outside of the object definition block and make them consistent.

File: wfl-slack-app/src/handle-wfl-request.js
Comment (JChapman202) [reply wfl-slack-app/src/handle-wfl-request.js:72]: we should discuss promises a bit further at some point, just for awareness purposes.  There is a little bit of callback hell here, but as projects get more and more complex you tend to see more and more of these sorts of nesting callbacks.

**Pull request merged**
